includes/class-wc-payment-gateways.php

```diff
@@ -152,7 +152,7 @@ public function get_available_payment_gateways() {
 
 		foreach ( $this->payment_gateways as $gateway ) {
 			if ( $gateway->is_available() ) {
-				if ( ! is_add_payment_method_page() ) {
+				if ( ! is_add_payment_method_page() && ! is_payment_methods_page() ) {
```

Comment:
Can you perhaps instead of adding a new function just amend the is_add_payment_methods_page function to also check for the payment-methods query var with an OR?